activerecord/test/cases/tasks/database_tasks_test.rb

```diff
@@ -42,11 +42,17 @@ def with_stubbed_new(&block)
 
   module DatabaseTasksHelper
     def assert_called_for_configs(method_name, configs, &block)
-      mock = Minitest::Mock.new
-      configs.each { |config| mock.expect(:call, nil, config) }
+      with_stubbed_temporary_connection do
+        mock = Minitest::Mock.new
+        configs.each { |config| mock.expect(:call, nil, config) }
 
-      ActiveRecord::Tasks::DatabaseTasks.stub(method_name, mock, &block)
-      assert_mock(mock)
+        ActiveRecord::Tasks::DatabaseTasks.stub(method_name, mock, &block)
+        assert_mock(mock)
+      end
+    end
+
+    def with_stubbed_temporary_connection(&block)
+      ActiveRecord::Tasks::DatabaseTasks.stub(:with_temporary_connection, ->(*, &block) { block.call }, &block)
```

Comment:
Just passes through with no asserting. Didn't think it was important since the logic doesn't pertain to it. 

I did build a version with some asserting (the configs are passed to it) so if that's what we want ... I can swap it out.